sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -268,7 +270,9 @@ case class CatalogTable(
    * schema of this table's data columns
    */
   def dataSchema: StructType = {
-    val dataFields = schema.dropRight(partitionColumnNames.length)
+    val dataFields = schema.filterNot { i =>
```

Comment:
`i` is easy to associate with `index`. Should we change this variable name?

Reply:
ok. I will change it. Thank you.